File: pkg/settings/spec/consts.go
package spec

import (
	aiproviderSpec "github.com/flexigpt/flexiui/pkg/aiprovider/spec"
)

// Define the default AI settings
var DefaultAISettings = map[aiproviderSpec.ProviderName]AISetting{
	aiproviderSpec.ANTHROPIC: {
		ApiKey:             aiproviderSpec.AnthropicProviderInfo.ApiKey,
		DefaultModel:       aiproviderSpec.AnthropicProviderInfo.DefaultModel,
		DefaultOrigin:      aiproviderSpec.AnthropicProviderInfo.DefaultOrigin,
		DefaultTemperature: aiproviderSpec.AnthropicProviderInfo.DefaultTemperature,
		AdditionalSettings: aiproviderSpec.AnthropicProviderInfo.AdditionalSettings,
	},
	aiproviderSpec.GOOGLE: {
		ApiKey:             aiproviderSpec.GoogleProviderInfo.ApiKey,
		DefaultModel:       aiproviderSpec.GoogleProviderInfo.DefaultModel,
		DefaultOrigin:      aiproviderSpec.GoogleProviderInfo.DefaultOrigin,
		DefaultTemperature: aiproviderSpec.GoogleProviderInfo.DefaultTemperature,
		AdditionalSettings: aiproviderSpec.GoogleProviderInfo.AdditionalSettings,
	},
	aiproviderSpec.HUGGINGFACE: {
		ApiKey:             aiproviderSpec.HuggingfaceProviderInfo.ApiKey,
		DefaultModel:       aiproviderSpec.HuggingfaceProviderInfo.DefaultModel,
		DefaultOrigin:      aiproviderSpec.HuggingfaceProviderInfo.DefaultOrigin,
		DefaultTemperature: aiproviderSpec.HuggingfaceProviderInfo.DefaultTemperature,
		AdditionalSettings: aiproviderSpec.HuggingfaceProviderInfo.AdditionalSettings,
	},
	aiproviderSpec.LLAMACPP: {
		ApiKey:             aiproviderSpec.LlamacppProviderInfo.ApiKey,
		DefaultModel:       aiproviderSpec.LlamacppProviderInfo.DefaultModel,
		DefaultOrigin:      aiproviderSpec.LlamacppProviderInfo.DefaultOrigin,
		DefaultTemperature: aiproviderSpec.LlamacppProviderInfo.DefaultTemperature,
		AdditionalSettings: aiproviderSpec.LlamacppProviderInfo.AdditionalSettings,
	},
	aiproviderSpec.OPENAI: {
		ApiKey:             aiproviderSpec.OpenaiProviderInfo.ApiKey,
		DefaultModel:       aiproviderSpec.OpenaiProviderInfo.DefaultModel,
		DefaultOrigin:      aiproviderSpec.OpenaiProviderInfo.DefaultOrigin,
		DefaultTemperature: aiproviderSpec.OpenaiProviderInfo.DefaultTemperature,
		AdditionalSettings: aiproviderSpec.OpenaiProviderInfo.AdditionalSettings,
	},
}

// Define the default settings data
var DefaultSettingsData = SettingsSchema{
	App: AppSettings{
		DefaultProvider: aiproviderSpec.OPENAI,
	},
	AISettings: AISettingsSchema{
		Anthropic:   DefaultAISettings[aiproviderSpec.ANTHROPIC],
		Google:      DefaultAISettings[aiproviderSpec.GOOGLE],
		Huggingface: DefaultAISettings[aiproviderSpec.HUGGINGFACE],
		Llamacpp:    DefaultAISettings[aiproviderSpec.LLAMACPP],
		Openai:      DefaultAISettings[aiproviderSpec.OPENAI]},
}

// Define the sensitive keys
var SensitiveKeys = []string{
	"openai.apiKey",
	"anthropic.apiKey",
	"huggingface.apiKey",
	"google.apiKey",
	"llamacpp.apiKey",
}

File: packages/settingstore/consts.ts
import {
	anthropicProviderInfo,
	googleProviderInfo,
	huggingfaceProviderInfo,
	llamacppProviderInfo,
	openaiProviderInfo,
	ProviderName,
} from 'aiprovider';
import { SettingsSchema } from './types';

export const defaultAISettings = {
	[ProviderName.ANTHROPIC]: {
		apiKey: anthropicProviderInfo.apiKey,
		defaultModel: anthropicProviderInfo.defaultModel,
		defaultOrigin: anthropicProviderInfo.defaultOrigin,
		defaultTemperature: anthropicProviderInfo.defaultTemperature,
		additionalSettings: anthropicProviderInfo.additionalSettings,
	},
	[ProviderName.GOOGLE]: {
		apiKey: googleProviderInfo.apiKey,
		defaultModel: googleProviderInfo.defaultModel,
		defaultOrigin: googleProviderInfo.defaultOrigin,
		defaultTemperature: googleProviderInfo.defaultTemperature,
		additionalSettings: googleProviderInfo.additionalSettings,
	},
	[ProviderName.HUGGINGFACE]: {
		apiKey: huggingfaceProviderInfo.apiKey,
		defaultModel: huggingfaceProviderInfo.defaultModel,
		defaultOrigin: huggingfaceProviderInfo.defaultOrigin,
		defaultTemperature: huggingfaceProviderInfo.defaultTemperature,
		additionalSettings: huggingfaceProviderInfo.additionalSettings,
	},
	[ProviderName.LLAMACPP]: {
		apiKey: llamacppProviderInfo.apiKey,
		defaultModel: llamacppProviderInfo.defaultModel,
		defaultOrigin: llamacppProviderInfo.defaultOrigin,
		defaultTemperature: llamacppProviderInfo.defaultTemperature,
		additionalSettings: llamacppProviderInfo.additionalSettings,
	},
	[ProviderName.OPENAI]: {
		apiKey: openaiProviderInfo.apiKey,
		defaultModel: openaiProviderInfo.defaultModel,
		defaultOrigin: openaiProviderInfo.defaultOrigin,
		defaultTemperature: openaiProviderInfo.defaultTemperature,
		additionalSettings: openaiProviderInfo.additionalSettings,
	},
};

export const defaultSettingsData: SettingsSchema = {
	app: {
		defaultProvider: ProviderName.OPENAI,
	},
	aiSettings: defaultAISettings,
};

export const sensitiveKeys = [
	'openai.apiKey',
	'anthropic.apiKey',
	'huggingface.apiKey',
	'google.apiKey',
	'llamacpp.apiKey',
];
